web_disable_export_group/static/src/js/list_controller.esm.js

```diff
@@ -1,20 +1,23 @@
 /** @odoo-module **/
 /* Copyright 2018 Tecnativa - David Vidal
    License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl). */
-import {ListController} from "@web/views/list/list_controller";
-import {onWillStart} from "@odoo/owl";
+import {archParseBoolean} from "@web/views/utils";
+import {exportAllItem} from "@web/views/list/export_all/export_all";
 import {patch} from "@web/core/utils/patch";
 
-patch(ListController.prototype, "disable_export_group", {
-    setup() {
-        this._super(...arguments);
-        onWillStart(async () => {
-            this.isExportEnable = await this.userService.hasGroup(
-                "base.group_allow_export"
-            );
-            this.isExportXlsEnable = await this.userService.hasGroup(
+patch(exportAllItem, {
+    async isDisplayed(env) {
+        const ExportEnable = super.isDisplayed(...arguments);
+        const ExportXlsEnable =
+            env.config.viewType === "list" &&
+            !env.model.root.selection.length &&
+            (await env.model.user.hasGroup(
                 "web_disable_export_group.group_export_xlsx_data"
-            );
-        });
+            )) &&
+            archParseBoolean(env.config.viewArch.getAttribute("export_xlsx"), true);
+        if (ExportXlsEnable) {
+            return ExportXlsEnable;
+        }
+        return ExportEnable;
```

Comment:
It's the same:
```suggestion
        return ExportXlsEnable || ExportEnable ;
```

On the picky side, the usual convention is that local variables are declared with lower_case (snake_case) (e.g: `export_xls_enable`) and PascalCase is usually used for classes

Reply:
@chienandalu 
done the requested changes